opentelemetry-appender-tracing/src/layer.rs

```diff
@@ -88,15 +88,14 @@ impl<LR: LogRecord> tracing::field::Visit for EventVisitor<'_, LR> {
         if is_duplicated_metadata(field.name()) {
             return;
         }
-        //TODO: Consider special casing "message" to populate body and document
-        // to users to use message field for log message, to avoid going to the
-        // record_debug, which has dyn dispatch, string allocation and
-        // formatting cost.
-
         //TODO: Fix heap allocation. Check if lifetime of &str can be used
         // to optimize sync exporter scenario.
-        self.log_record
-            .add_attribute(Key::new(field.name()), AnyValue::from(value.to_owned()));
+        if field.name() == "message" {
```

Comment:
This would be a breaking change if user is expecting the message to go an attribute?

Reply:
> This would be a breaking change if user is expecting the message to go an attribute?

There was no contract documented about this, so while this change behavior, I don't think it should be a major issue. (We were already special casing "message" for the record_debug, so missing this may also be considered a bug)

Reply:
> And what should be the expected behavior if both message attribute and default body field is provided

We can add tests to make sure this is covered, but the behavior is really controlled by tracing itself. We can make some decision and document it in the event of conflict, as there can only be one body.

Reply:
https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-appender-tracing/src/layer.rs#L121 Fixing these TODOs are also going to change the behavior, so is fixing error to be reported specially. It's questionable to that 1.0 (but maybe okay), but I don't see an issue doing it at this stage. Let me know your thoughts.

Reply:
We need to document the final behavior before 1.0.
https://github.com/open-telemetry/opentelemetry-rust/issues/2150 will also be related to this...